src/DigitalGardenSiteManager.ts

```diff
@@ -10,22 +10,13 @@ import { Octokit } from "@octokit/core";
 import { Base64 } from "js-base64";
 import type DigitalGardenPluginInfo from "./DigitalGardenPluginInfo";
 
-export interface IDigitalGardenSiteManager {
-	getNoteUrl: (file: TFile) => string;
-	getNoteHashes: () => Promise<Record<string, string>>;
-	getImageHashes: () => Promise<Record<string, string>>;
-	createPullRequestWithSiteChanges: () => Promise<string>;
-}
```

Comment:
This one's up to you! These are typed in the class too, and not sure what pattern you want to follow with these. 

If these were abstracted / implemented more this would make more sense! 

(i'm mostly a fp-ts kinda guy, so trust your vision on class stuff)

Reply:
In my day-to-day work I usually code in C# and dotnet. A very common pattern there is to create interfaces for integration classes, that is classes talking to the outside world, so that we can use Dependency Injection for the classes that are using it and thus easily be able to unit test any consumers of this class. 

That was my initial thought for this, making it easy to write unit tests. But as you know I never got around to write the unit tests, and thus all the interfaces aren't really of any help in the current codebase. 

I'll accept this PR with this deleted for now. Whenever we create a unit test for a class where this is used it is very easy to rewrite as an interface again.

Reply:
Cool! I definitely see the value with abstractions like modals and reusable things. It's been 10 years since I've touched Java so it's been a while. I'm for types and abstractions as long as it's consistent and helps with readability. Need to wrap my head around where it's useful in cases like these, still a lot of things with js class patterns that are a little hazy for me